src/endpoints/pool/pool.service.ts

```diff
@@ -0,0 +1,113 @@
+import { GatewayService } from "src/common/gateway/gateway.service";
+import { ApiConfigService } from "src/common/api-config/api.config.service";
+import { CacheService } from "@multiversx/sdk-nestjs-cache";
+import { CacheInfo } from "src/utils/cache.info";
+import { TxInPoolFields, TxPoolGatewayResponse } from "src/common/gateway/entities/transaction.pool";
+import { TransactionType } from "../transactions/entities/transaction.type";
+import { TransactionInPool } from "./entities/transaction.in.pool.dto";
+import { PoolFilter } from "./entities/pool.filter";
+
+@Injectable()
+export class PoolService {
+  constructor(
+    private readonly gatewayService: GatewayService,
+    private readonly apiConfigService: ApiConfigService,
+    private readonly cacheService: CacheService,
+  ) { }
+
+  async getTransactionFromPool(txHash: string): Promise<TransactionInPool | undefined> {
+    const pool = await this.getEntirePool();
+    const transaction = pool.find(tx => tx.txHash === txHash);
+
+    return transaction;
+  }
+
+  async getPoolCount(filter: PoolFilter): Promise<number> {
+    const pool = await this.getEntirePool();
+    return this.applyFilters(pool, filter).length;
+  }
+
+  async getPool(
+    queryPagination: QueryPagination,
+    filter: PoolFilter,
+  ): Promise<TransactionInPool[]> {
+    if (!this.apiConfigService.isTransactionPoolEnabled()) {
+      return [];
+    }
+
+    const { from, size } = queryPagination;
+    const entirePool = await this.getEntirePool();
+    const pool = this.applyFilters(entirePool, filter);
+    return pool.slice(from, from + size);
+  }
+
+  async getEntirePool(): Promise<TransactionInPool[]> {
+    return await this.cacheService.getOrSet(
+      CacheInfo.TransactionPool.key,
+      async () => await this.getTxPoolRaw(),
+      CacheInfo.TransactionPool.ttl
+    );
+  }
+
+  async getTxPoolRaw(): Promise<TransactionInPool[]> {
+    const pool = await this.gatewayService.getTransactionPool();
+    return this.parseTransactions(pool);
+  }
+
+  private parseTransactions(rawPool: TxPoolGatewayResponse): TransactionInPool[] {
+    const transactionPool: TransactionInPool[] = [];
+    if (rawPool && rawPool.txPool) {
+      transactionPool.push(...this.processTransactionType(rawPool.txPool.regularTransactions || [], TransactionType.Transaction));
+      transactionPool.push(...this.processTransactionType(rawPool.txPool.smartContractResults || [], TransactionType.SmartContractResult));
+      transactionPool.push(...this.processTransactionType(rawPool.txPool.rewards || [], TransactionType.Reward));
+    }
+
+    return transactionPool;
+  }
+
+  private processTransactionType(transactions: any[], transactionType: TransactionType): TransactionInPool[] {
+    return transactions.map(tx => this.parseTransaction(tx.txFields, transactionType));
+  }
+
+  private parseTransaction(tx: TxInPoolFields, type: TransactionType): TransactionInPool {
+    return new TransactionInPool({
+      txHash: tx.hash || '',
```

Comment:
similar to above, use the preferred `??` operator

Reply:
done